internal/featuretests/v3/upstreamtls_test.go

```diff
@@ -454,3 +454,156 @@ func TestBackendTLSPolicyPrecedenceOverUpstreamProtocolAnnotationWithHTTPRoute(t
 		TypeUrl: clusterType,
 	})
 }
+
```

Comment:
A small comment describing why we're adding this test would make this complete for me! (something to the effect of when there is an HTTPProxy with upstream tls settings and a HTTPRoute+BackendTLSPolicy and they all point to the same service, if there is a difference in TLS settings then distinct clusters are generated)